src/server/transaction.cc

```diff
@@ -276,8 +259,8 @@ void Transaction::InitByKeys(KeyIndex key_index) {
   // Initialize shard data based on distributed arguments.
   InitShardData(shard_index, key_index.num_args(), needs_reverse_mapping);
 
-  if (multi_)
-    InitMultiData(key_index);
+  if (multi_ && !multi_->lock_mode)
```

Comment:
added `lock_mode` variable being determined at the lock time because `Mode()` is changing according to the command being set within the multi transaction.